tests/unit/random_tests.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import collections
+import unittest
+
+from baseplate import random
+
+from .. import mock
+
+
+class WeightedChoiceTests(unittest.TestCase):
+    def test_empty_list(self):
+        with self.assertRaises(ValueError):
+            random.weighted_choice([], lambda i: i)
+
+    def test_negative_weight(self):
+        with self.assertRaises(ValueError):
+            random.weighted_choice(["a"], lambda i: -1)
+
+    @mock.patch("random.random")
+    def test_iterator(self, random_random):
```

Comment:
Maybe use `patched_random` here?  It's what I tend to see in the `mock` docs, iirc.